src/main/java/com/internet/shop/security/AuthenticationServiceImpl.java

```diff
@@ -15,7 +16,8 @@
     @Override
     public User login(String login, String password) throws AuthenticationException {
         Optional<User> user = userService.findByLogin(login);
-        if (user.isEmpty() || !user.get().getPassword().equals(password)) {
+        if (user.isEmpty() || !HashUtil.hashPassword(password, user.get().getSalt())
+                        .equals(user.get().getPassword())) {
```

Comment:
```suggestion
        if (user.isEmpty() || !isPasswordValid(user.get(), password)) {
```
let's create a private method here with this logic